File: 1127-last-stone-weight/1127-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        if len(stones) == 1:
            return stones[0]
        for i in range(len(stones)):
            stones[i] = stones[i] * -1
        heapq.heapify(stones)
        while(len(stones) > 1):
            a = -1 * heapq.heappop(stones)
            b = -1* heapq.heappop(stones)
            if(abs(a - b)):
                heapq.heappush(stones, -1 * abs(a-b))
        if stones:
            return -1 * stones[0]
        return 0

File: 1127-last-stone-weight/1127-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {

        if(stones.length == 1){
            return stones[0];
        }
        PriorityQueue<Integer> p = new PriorityQueue<>();
        for(int i = 0; i < stones.length; i++){
            p.add( stones[i] * -1);
            
        }
        int a = 0;
        int b = 0;
         
        
        while(p.size() > 1){
            a = -1 * p.poll();
            b = -1 * p.poll();

            if(Math.abs(a - b) > 0){
                p.add(-1 * Math.abs(a-b));
            }
        }
        if(p.size() > 0){
            return -1 * p.peek();
        }
        return 0;


    }
}